explorer/src/context/hooks.ts

```diff
@@ -1,30 +1,46 @@
 import * as React from 'react';
 import { ApiState } from '../typeDefs/explorer-api';
 
-type WrappedApiFn<T> = () => Promise<ApiState<T>>;
-
+/**
+ * Custom hook to get data from the API by passing an id to a delegate method that fetches the data asynchronously
+ * @param id                The id to fetch
+ * @param fn                Delegate the fetching to this method (must take `(id: string)` as a parameter)
+ * @param errorMessage      A static error message, to use when no dynamic error message is returned
+ */
 export const useApiState = <T>(
-  // eslint-disable-next-line @typescript-eslint/ban-types
-  fn: Function,
+  id: string,
+  fn: (argId: string) => Promise<T>,
   errorMessage: string,
-): [ApiState<T>, WrappedApiFn<T>] => {
-  const [value, setValue] = React.useState<ApiState<T>>();
-  const wrappedFn = React.useCallback(async () => {
+): [ApiState<T> | undefined, () => Promise<ApiState<T>>, () => void] => {
+  // stores the state
+  const [value, setValue] = React.useState<ApiState<T> | undefined>();
+
+  // clear the value
+  const clearValueFn = () => setValue(undefined);
+
+  // this provides a method to trigger the delegate to fetch data
+  const wrappedFetchFn = React.useCallback(async () => {
     try {
+      // keep previous state and set to loading
       setValue((prevState) => ({ ...prevState, isLoading: true }));
-      const data = await fn();
-      setValue({
+
+      // delegate to user function to get data and set if successful
+      const data = await fn(id);
+      const newValue: ApiState<T> = {
         isLoading: false,
         data,
-      });
-      return data;
+      };
+      setValue(newValue);
```

Comment:
I think this could just be
```
setValue({
  isLoading: false,
  data
})
```

Reply:
@fmtabbara, do you mean by not returning the newValue object after set it? the function is expected to have a return value and the logic doesn't work if it doesn't.

Reply:
I just thought instead of saving the object into a variable and setting that variable in state, just set the object directly to state?

Reply:
I see what you mean! Ignore me.